### ISSUE TITLE
FAMD: Performance Warning DataFrame is highly fragmented

### ISSUE THREAD

**Issue by CH4LLENG3R** [MaxHalford/prince#208]
prince\famd.py:75: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  eta2[col] = (

Warning is caused probably due to using a newer version of pandas with prince.

**Comment by MaxHalford** [comment]
Thanks for opening the issue. I'll take a look in a couple of weeks. In the meantime, can you share more info about your setup so that I reproduce the warning?

**Comment by CH4LLENG3R** [comment]
Replacing eta[col] to eta.loc[:, col] should solve the issue.

I use Python 3.10.8 and the pip requirements are as follows:
[requirements.txt](https://github.com/user-attachments/files/26888978/requirements.txt)

Here is the code snippet where I call the famd:
```py
def reduce_famd(x, variance_threshold=0.9, initial_guess=500):
    """
    Applies FAMD in a single pass. 
    Returns components hitting the threshold, or all calculated components if threshold isn't met.
    """
    # Total Inertia for mixed data: 
    # (Continuous count * 1) + (Sum of (levels - 1) for each categorical)
    # Since yours are binary (2 levels), levels - 1 = 1. So it's just x.shape[1].
    total_inertia = x.shape[1]
    
    print(f"Running FAMD with n_components={initial_guess}...")
    famd = FAMD(
        n_components=initial_guess,
        n_iter=5,
        copy=True,
        check_input=True,
        engine='sklearn',
        random_state=42 # Or your SEED
    )
    
    # Fit and transform in one go to be efficient
    x_transformed = famd.fit_transform(x)
    
    # Calculate cumulative variance ratio
    # prince.eigenvalues_ gives the inertia explained by each component
    explained_vars = np.array(famd.eigenvalues_) / total_inertia
    cumsum = np.cumsum(explained_vars)
    
    # Check if we hit the threshold
    reached_threshold = cumsum >= variance_threshold
    
    if np.any(reached_threshold):
        n_needed = np.argmax(reached_threshold) + 1
        print(f"Threshold reached! Using {n_needed} components for {cumsum[n_needed-1]:.2%} variance.")
        # Slice the DataFrame/Array to the required components
        x_reduced = x_transformed.iloc[:, :n_needed].values
    else:
        print(f"Threshold NOT reached with {initial_guess} components.")
        print(f"Total variance explained: {cumsum[-1]:.2%}. Returning all components.")
        x_reduced = x_transformed.values
        
    return x_reduced
```

**Comment by MaxHalford** [comment]
Thanks for the additional context. I wasn't able to reproduce the issue. However I shipped a new version (0.17.1) which should resolve the issue, following the `eta` change you point out. Could you try it out and see if it removes the warning?
